File: src/com/l2kt/gameserver/network/clientpackets/RequestCursedWeaponLocation.java
package com.l2kt.gameserver.network.clientpackets;

import java.util.ArrayList;
import java.util.List;

import com.l2kt.gameserver.data.manager.CursedWeaponManager;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.model.entity.CursedWeapon;
import com.l2kt.gameserver.model.location.Location;
import com.l2kt.gameserver.network.serverpackets.ExCursedWeaponLocation;
import com.l2kt.gameserver.network.serverpackets.ExCursedWeaponLocation.CursedWeaponInfo;

public final class RequestCursedWeaponLocation extends L2GameClientPacket
{
	@Override
	protected void readImpl()
	{
	}
	
	@Override
	protected void runImpl()
	{
		final Player player = getClient().getActiveChar();
		if (player == null)
			return;
		
		final List<CursedWeaponInfo> list = new ArrayList<>();
		for (CursedWeapon cw : CursedWeaponManager.getInstance().getCursedWeapons())
		{
			if (!cw.isActive())
				continue;
			
			final Location loc = cw.getWorldPosition();
			if (loc != null)
				list.add(new CursedWeaponInfo(loc, cw.getItemId(), (cw.isActivated()) ? 1 : 0));
		}
		
		if (!list.isEmpty())
			player.sendPacket(new ExCursedWeaponLocation(list));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestCursedWeaponLocation.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.data.manager.CursedWeaponManager
import com.l2kt.gameserver.network.serverpackets.ExCursedWeaponLocation
import com.l2kt.gameserver.network.serverpackets.ExCursedWeaponLocation.CursedWeaponInfo
import java.util.*

class RequestCursedWeaponLocation : L2GameClientPacket() {
    override fun readImpl() {}

    override fun runImpl() {
        val player = client.activeChar ?: return

        val list = ArrayList<CursedWeaponInfo>()
        for (cw in CursedWeaponManager.cursedWeapons) {
            if (!cw.isActive)
                continue

            val loc = cw.worldPosition
            if (loc != null)
                list.add(CursedWeaponInfo(loc, cw.itemId, if (cw.isActivated) 1 else 0))
        }

        if (!list.isEmpty())
            player.sendPacket(ExCursedWeaponLocation(list))
    }
}